pkg/plugin/system/system_test.go

```diff
@@ -0,0 +1,252 @@
+/*
+Copyright © 2022 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package system
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+
+	rootFlags "aspect.build/cli/pkg/aspect/root/flags"
+	"aspect.build/cli/pkg/aspecterrors"
+	"aspect.build/cli/pkg/ioutils"
+	plugin_mock "aspect.build/cli/pkg/plugin/sdk/v1alpha2/plugin/mock"
+)
+
+func TestPluginSystem(t *testing.T) {
+	t.Run("executes hooks in reverse order of interceptors", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		// Minimal streams
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		// empty ctx + args + command
+		ctx := context.Background()
+		args := []string{}
+		cmd := &cobra.Command{
+			Use: "HookOrderTest",
+		}
+		// Required flags for interceptor hooks
+		cmd.PersistentFlags().Bool(rootFlags.InteractiveFlagName, false, "")
+
+		plugin := plugin_mock.NewMockPlugin(ctrl)
+
+		// Expect the callbacks in reverse-order of execution
+		gomock.InOrder(
+			plugin.EXPECT().PostRunHook(gomock.Any(), gomock.Any()),
+			plugin.EXPECT().PostTestHook(gomock.Any(), gomock.Any()),
+			plugin.EXPECT().PostBuildHook(gomock.Any(), gomock.Any()),
+		)
+
+		ps := NewPluginSystem().(*pluginSystem)
+		ps.Configure(streams)
```

Comment:
I think by doing this you are being too broad in your tests. Try to separate better in unit tests vs integration tests. I.e. if you are using the constructed plugin system from `NewPluginSystem`, you are already being too broad. E.g. `Configure` should have unit tests and the cases should be something like:

- Configure fails when Finder fails.
- Configure fails when Parser fails.
- Configure fails when getting an RPC Client fails (this one will be trickier and you will need to return a Client mock from a ClientFactory mock).
- Configure fails when getting a raw Client fails (same here).

Reply:
I'm trying to unit test `commandHooksInterceptor`, I really don't care about `Configure` and would rather not have to invoke this at all. `Configure` should have its own tests like you're saying.

I guess that's what I don't like about the "repeated setup" that I mentioned earlier... it seems like too much and too fragile for a unit test so I'd like to either remove it or put it in a single place.

I think we're saying the same thing though...